lightkurve/correctors/designmatrix.py

```diff
@@ -224,12 +239,19 @@ def append_constant(self, prior_mu=0, prior_sigma=np.inf):
             New design matrix with a column of ones appended. This column is
             named "offset".
         """
-        extra_df = pd.DataFrame(np.atleast_2d(np.ones(self.shape[0])).T, columns=['offset'])
-        new_df = pd.concat([self.df, extra_df], axis=1)
-        prior_mu = np.append(self.prior_mu, prior_mu)
-        prior_sigma = np.append(self.prior_sigma, prior_sigma)
-        return DesignMatrix(new_df, name=self.name,
-                            prior_mu=prior_mu, prior_sigma=prior_sigma)
+        if inplace:
+            self.df.insert(self.df.shape[1], 'offset', 1, allow_duplicates=False)
+            self.prior_mu = np.append(self.prior_mu, prior_mu)
+            self.prior_sigma = np.append(self.prior_sigma, prior_sigma)
+            if self._sparse:
+                self._sparse_values = hstack([self.values, csr_matrix(np.ones(self.values.shape[0])).T], format='csr')
+            return self
+        else:
+            df = self.df.copy()
+            df.insert(self.df.shape[1], 'offset', 1, allow_duplicates=False)
+            return DesignMatrix(df, prior_mu=np.append(self.prior_mu, prior_mu),
+                                prior_sigma=np.append(self.prior_sigma, prior_sigma), name=self.name,
+                                sparse=self._sparse)
```

Comment:
There's quite a lot of code duplication in the function above.  Could we refactor it like this:
```python
if inplace:
    dm = self
else:
    dm = self.copy()
# insert pca logic here
return dm
```

Reply:
Yeah this is great, I like it.